rotkehlchen/chain/evm/active_management/manager.py

```diff
@@ -0,0 +1,153 @@
+import logging
+from collections.abc import Callable
+from typing import TYPE_CHECKING, Any
+
+from web3 import Web3
+from web3.exceptions import Web3Exception
+from web3.types import Wei
+
+from rotkehlchen.chain.ethereum.utils import asset_raw_value
+from rotkehlchen.errors.misc import RemoteError
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+
+if TYPE_CHECKING:
+    from rotkehlchen.assets.asset import EvmToken
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+    from rotkehlchen.chain.evm.types import Web3Node
+    from rotkehlchen.types import ChecksumEvmAddress, FVal
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class ActiveManager:
+
+    def __init__(self, node_inquirer: 'EvmNodeInquirer'):
+        self.node_inquirer = node_inquirer
+
+    def addresses_interacted(
+            self,
+            from_address: 'ChecksumEvmAddress',
+            to_address: 'ChecksumEvmAddress',
+    ) -> bool:
+        """Check if from address interacted before with to_address"""
+        with self.node_inquirer.database.conn.read_ctx() as cursor:
+            cursor.execute(
+                'SELECT COUNT(*) FROM history_events JOIN evm_events_info ON '
+                'history_events.identifier=evm_events_info.identifier WHERE '
+                'location_label=? AND address=?',
+                (from_address, to_address),
+            )
+            return cursor.fetchone()[0] > 0
+
+    def _query_gas(self, web3: Web3) -> dict[str, int]:
```

Comment:
etherscan also provides endpoints to query gas 
- https://docs.etherscan.io/api-endpoints/gas-tracker#get-gas-oracle
- https://docs.etherscan.io/api-endpoints/geth-parity-proxy#eth_estimategas

there's no privacy concern with this i guess.

Reply:
for this I believe it is better to keep web3 nodes for now and we can check to use etherscan later